### ISSUE TITLE
Windows 2016 VM failed to get IP from DHCP server

### ISSUE THREAD

**Issue by cynepco3hahue** [kubevirt/kubevirt#789]
### Description
VM with Windows 2016 failed to get IP from DHCP server
### VM details
```
apiVersion: kubevirt.io/v1alpha1
kind: VirtualMachine
metadata:
  name: windows2016
spec:
  terminationGracePeriodSeconds: 0
  domain:
    cpu:
      cores: 2
    features:
      acpi: {}
      apic: {}
      hyperv:
        relaxed: {}
        vapic: {}
        spinlocks:
          spinlocks: 8191
    clock:
      utc: {}
      timer:
        hpet:
          present: false
        pit:
          tickPolicy: delay
        rtc:
          tickPolicy: catchup
        hyperv: {}
    firmware:
      uuid: 5d307ca9-b3ef-428c-8861-06e72d69f223
    resources:
      requests:
        memory: 4096M
    devices:
      disks:
      - name: server2016
        volumeName: server2016
        disk:
          dev: vda
  volumes:
    - name: server2016
      persistentVolumeClaim:
        claimName: disk-windows
```
### VM libvirt xml
```
# virsh -r dumpxml 1
<domain type='qemu' id='1'>
  <name>default_windows2016</name>
  <uuid>5d307ca9-b3ef-428c-8861-06e72d69f223</uuid>
  <metadata>
    <kubevirt xmlns="http://kubevirt.io">
      <uid>e8232248-1f90-11e8-9eca-525400c99dab</uid>
      <graceperiod>
        <deletionGracePeriodSeconds>0</deletionGracePeriodSeconds>
      </graceperiod>
    </kubevirt>
  </metadata>
  <memory unit='KiB'>4000768</memory>
  <currentMemory unit='KiB'>4000000</currentMemory>
  <vcpu placement='static'>2</vcpu>
  <resource>
    <partition>/machine</partition>
  </resource>
  <sysinfo type='smbios'>
    <system>
      <entry name='uuid'>5d307ca9-b3ef-428c-8861-06e72d69f223</entry>
    </system>
  </sysinfo>
  <os>
    <type arch='x86_64' machine='pc-q35-2.10'>hvm</type>
    <boot dev='hd'/>
  </os>
  <features>
    <acpi/>
    <apic/>
    <hyperv>
      <relaxed state='on'/>
      <vapic state='on'/>
      <spinlocks state='on' retries='8191'/>
    </hyperv>
  </features>
  <cpu>
    <topology sockets='1' cores='2' threads='1'/>
  </cpu>
  <clock offset='utc' adjustment='reset'>
    <timer name='rtc' present='yes' tickpolicy='catchup'/>
    <timer name='pit' present='yes' tickpolicy='delay'/>
    <timer name='hpet' present='no'/>
    <timer name='hypervclock' present='yes'/>
  </clock>
  <on_poweroff>destroy</on_poweroff>
  <on_reboot>restart</on_reboot>
  <on_crash>destroy</on_crash>
  <devices>
    <emulator>/usr/bin/qemu-system-x86_64</emulator>
    <disk type='file' device='disk'>
      <driver name='qemu' type='raw'/>
      <source file='/var/run/kubevirt-private/vm-disks/server2016/disk.img'/>
      <backingStore/>
      <target dev='sda' bus='sata'/>
      <alias name='sata0-0-0'/>
      <address type='drive' controller='0' bus='0' target='0' unit='0'/>
    </disk>
    <disk type='file' device='cdrom'>
      <driver name='qemu' type='raw'/>
      <source file='/var/run/kubevirt-private/vm-disks/tools/disk.img'/>
      <backingStore/>
      <target dev='sdb' bus='sata'/>
      <readonly/>
      <alias name='sata0-0-1'/>
      <address type='drive' controller='0' bus='0' target='0' unit='1'/>
    </disk>
    <controller type='usb' index='0' model='nec-xhci'>
      <alias name='usb'/>
      <address type='pci' domain='0x0000' bus='0x02' slot='0x00' function='0x0'/>
    </controller>
    <controller type='sata' index='0'>
      <alias name='ide'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1f' function='0x2'/>
    </controller>
    <controller type='pci' index='0' model='pcie-root'>
      <alias name='pcie.0'/>
    </controller>
    <controller type='pci' index='1' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='1' port='0x10'/>
      <alias name='pci.1'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x0' multifunction='on'/>
    </controller>
    <controller type='pci' index='2' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='2' port='0x11'/>
      <alias name='pci.2'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x1'/>
    </controller>
    <controller type='pci' index='3' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='3' port='0x12'/>
      <alias name='pci.3'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x2'/>
    </controller>
    <controller type='pci' index='4' model='pcie-root-port'>
      <model name='pcie-root-port'/>
      <target chassis='4' port='0x13'/>
      <alias name='pci.4'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x3'/>
    </controller>
    <interface type='bridge'>
      <mac address='0a:58:0a:80:00:44'/>
      <source bridge='br1'/>
      <target dev='vnet0'/>
      <model type='virtio'/>
      <alias name='net0'/>
      <address type='pci' domain='0x0000' bus='0x01' slot='0x00' function='0x0'/>
    </interface>
    <serial type='unix'>
      <source mode='bind' path='/var/run/kubevirt-private/default/windows2016/virt-serial0'/>
      <target port='0'/>
      <alias name='serial0'/>
    </serial>
    <console type='unix'>
      <source mode='bind' path='/var/run/kubevirt-private/default/windows2016/virt-serial0'/>
      <target type='serial' port='0'/>
      <alias name='serial0'/>
    </console>
    <input type='mouse' bus='ps2'>
      <alias name='input0'/>
    </input>
    <input type='keyboard' bus='ps2'>
      <alias name='input1'/>
    </input>
    <graphics type='vnc' socket='/var/run/kubevirt-private/default/windows2016/virt-vnc'>
      <listen type='socket' socket='/var/run/kubevirt-private/default/windows2016/virt-vnc'/>
    </graphics>
    <video>
      <model type='vga' vram='16384' heads='1' primary='yes'/>
      <alias name='video0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x0'/>
    </video>
    <memballoon model='virtio'>
      <alias name='balloon0'/>
      <address type='pci' domain='0x0000' bus='0x03' slot='0x00' function='0x0'/>
    </memballoon>
  </devices>
  <seclabel type='dynamic' model='dac' relabel='yes'>
    <label>+107:+107</label>
    <imagelabel>+107:+107</imagelabel>
  </seclabel>
</domain>
```
### VM OS network details
![windows_networking_problem](https://user-images.githubusercontent.com/4704810/36944826-505f08ba-1fac-11e8-8db6-34520c1bd933.png)
### Virt-launcher pod network
```
# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
3: eth0@if30: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue master br1 state UP group default 
    link/ether e2:0f:e8:0e:44:ab brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet6 fe80::e00f:e8ff:fe0e:44ab/64 scope link 
       valid_lft forever preferred_lft forever
4: br1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP group default 
    link/ether e2:0f:e8:0e:44:ab brd ff:ff:ff:ff:ff:ff
    inet 10.11.12.13/24 brd 10.11.12.255 scope global br1
       valid_lft forever preferred_lft forever
    inet6 fe80::e00f:e8ff:fe0e:44ab/64 scope link 
       valid_lft forever preferred_lft forever
5: vnet0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc pfifo_fast master br1 state UNKNOWN group default qlen 500
    link/ether fe:58:0a:80:00:44 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::fc58:aff:fe80:44/64 scope link 
       valid_lft forever preferred_lft forever
```

**Comment by cynepco3hahue** [comment]
@vladikr @rmohr FYI

**Comment by vladikr** [comment]
The problem is that Windows doesn't accept DHCP offers because our Server-ID and DNS options have length 16 and should be 4 bytes long for Windows to accept:
```
Your-IP 10.244.0.34                             
          Client-Ethernet-Address 0a:58:0a:f4:00:22                                                                                                                                                                
          Vendor-rfc1048 Extensions                                                                      
            Magic Cookie 0x63825363                                                                      
            DHCP-Message Option 53, length 1: Offer                                                                                                                                                                
            Server-ID Option 54, length 16: 0.0.0.0,0.0.0.0,0.0.255.255,10.11.12.13                      
            Lease-Time Option 51, length 4: 86313600                                                     
            Subnet-Mask Option 1, length 4: 255.255.255.0
            Default-Gateway Option 3, length 4: 10.244.0.1
            Domain-Name-Server Option 6, length 16: 0.0.0.0,0.0.0.0,0.0.255.255,8.8.8.8
            END Option 255, length 0
```

**Comment by fabiand** [comment]
The patch was merged, @cynepco3hahue do you have the opportunity to check if it's fixing the issue?

**Comment by cynepco3hahue** [comment]
The windows VM succeeds to get IP from DHCP server and I can ping `8.8.8.8`.
Fixed by #801
